AssertionError in rebindSuper

## Minimized code
The following compiles in a single package or removing protected modifier but fails in this form:
```Scala
package apackage {

  abstract class A {
    protected val x: Int
  }
  abstract class A2 {
    val x: Int
  }
}

package bpackage {
  import apackage._

  trait B extends A {
    println(x)
  }
  trait B2 extends A2 {
    println(x)
  }
}

package cpackage {
  import apackage._
  import bpackage._

  case class C(override protected val x: Int) extends A with B
  case class C2(override val x: Int) extends A2 with B2
}
```

## Output (click arrow to expand)
<details>

```scala
➜  snips ~/projects/dotty/bin/scalac -d /tmp i9329.scala
exception occurred while compiling i9329.scala
java.lang.AssertionError: assertion failed: cannot rebind method bpackage$B$$super$x, bpackage$B$$super$x x while compiling i9329.scala
Exception in thread "main" java.lang.AssertionError: assertion failed: cannot rebind method bpackage$B$$super$x, bpackage$B$$super$x x
        at scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:8)
        at dotty.tools.dotc.transform.ResolveSuper$.rebindSuper(ResolveSuper.scala:115)
        at dotty.tools.dotc.transform.ResolveSuper.superAccessors$2$$anonfun$2(ResolveSuper.scala:52)
        at scala.collection.immutable.List.map(List.scala:246)
        at dotty.tools.dotc.transform.ResolveSuper.superAccessors$3(ResolveSuper.scala:53)
        at dotty.tools.dotc.transform.ResolveSuper.$anonfun$1(ResolveSuper.scala:55)
        at scala.collection.immutable.List.flatMap(List.scala:293)
        at dotty.tools.dotc.transform.ResolveSuper.transformTemplate(ResolveSuper.scala:55)
        at dotty.tools.dotc.transform.ResolveSuper.transformTemplate(ResolveSuper.scala:43)
        at dotty.tools.dotc.transform.MegaPhase.goTemplate(MegaPhase.scala:1004)
        at dotty.tools.dotc.transform.MegaPhase.transformUnnamed$1(MegaPhase.scala:363)
        at dotty.tools.dotc.transform.MegaPhase.transformTree(MegaPhase.scala:429)
        at dotty.tools.dotc.transform.MegaPhase.transformNamed$1(MegaPhase.scala:256)
        at dotty.tools.dotc.transform.MegaPhase.transformTree(MegaPhase.scala:427)
        at dotty.tools.dotc.transform.MegaPhase.transformStat$2(MegaPhase.scala:437)
        at dotty.tools.dotc.transform.MegaPhase.recur$1(MegaPhase.scala:442)
        at dotty.tools.dotc.transform.MegaPhase.transformStats(MegaPhase.scala:442)
        at dotty.tools.dotc.transform.MegaPhase.mapPackage$1(MegaPhase.scala:382)
        at dotty.tools.dotc.transform.MegaPhase.transformUnnamed$1(MegaPhase.scala:385)
        at dotty.tools.dotc.transform.MegaPhase.transformTree(MegaPhase.scala:429)
        at dotty.tools.dotc.transform.MegaPhase.transformStat$2(MegaPhase.scala:439)
        at dotty.tools.dotc.transform.MegaPhase.recur$1(MegaPhase.scala:442)
        at dotty.tools.dotc.transform.MegaPhase.recur$1(MegaPhase.scala:442)
        at dotty.tools.dotc.transform.MegaPhase.recur$1(MegaPhase.scala:442)
        at dotty.tools.dotc.transform.MegaPhase.transformStats(MegaPhase.scala:442)
        at dotty.tools.dotc.transform.MegaPhase.mapPackage$1(MegaPhase.scala:382)
        at dotty.tools.dotc.transform.MegaPhase.transformUnnamed$1(MegaPhase.scala:385)
        at dotty.tools.dotc.transform.MegaPhase.transformTree(MegaPhase.scala:429)
        at dotty.tools.dotc.transform.MegaPhase.transformUnit(MegaPhase.scala:448)
        at dotty.tools.dotc.transform.MegaPhase.run(MegaPhase.scala:460)
        at dotty.tools.dotc.core.Phases$Phase.runOn$$anonfun$1(Phases.scala:300)
        at scala.collection.immutable.List.map(List.scala:246)
        at dotty.tools.dotc.core.Phases$Phase.runOn(Phases.scala:301)
        at dotty.tools.dotc.Run.runPhases$4$$anonfun$4(Run.scala:185)
        at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
        at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
        at scala.collection.ArrayOps$.foreach$extension(ArrayOps.scala:1323)
        at dotty.tools.dotc.Run.runPhases$5(Run.scala:195)
        at dotty.tools.dotc.Run.compileUnits$$anonfun$1(Run.scala:203)
        at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
        at dotty.tools.dotc.util.Stats$.maybeMonitored(Stats.scala:67)
        at dotty.tools.dotc.Run.compileUnits(Run.scala:210)
        at dotty.tools.dotc.Run.compileSources(Run.scala:146)
        at dotty.tools.dotc.Run.compile(Run.scala:130)
        at dotty.tools.dotc.Driver.doCompile(Driver.scala:39)
        at dotty.tools.dotc.Driver.process(Driver.scala:186)
        at dotty.tools.dotc.Driver.process(Driver.scala:155)
        at dotty.tools.dotc.Driver.process(Driver.scala:167)
        at dotty.tools.dotc.Driver.main(Driver.scala:194)
        at dotty.tools.dotc.Main.main(Main.scala)
```
</details>
Reported at scala/bug#12249

Split from #9329

--- a/compiler/src/dotty/tools/dotc/core/NameOps.scala
+++ b/compiler/src/dotty/tools/dotc/core/NameOps.scala
@@ -361,5 +361,14 @@ object NameOps {
       case raw.BANG  => UNARY_!
       case _ => name
     }
+
+    /** If this is a super accessor name, its underlying name, which is the name
+     *  of the method that the super accessor forwards to.
+     */
+    def originalOfSuperAccessorName: TermName = name match
+      case SuperAccessorName(name1)   => name1.originalOfSuperAccessorName
+      case ExpandedName(_, name1)     => name1.originalOfSuperAccessorName
+      case ExpandPrefixName(_, name1) => name1.originalOfSuperAccessorName
+      case _ => name
   }
 }

--- a/compiler/src/dotty/tools/dotc/transform/ResolveSuper.scala
+++ b/compiler/src/dotty/tools/dotc/transform/ResolveSuper.scala
@@ -118,7 +118,10 @@ object ResolveSuper {
           report.error(IllegalSuperAccessor(base, memberName, targetName, acc, accTp, other.symbol, otherTp), base.srcPos)
       bcs = bcs.tail
     }
-    assert(sym.exists, i"cannot rebind $acc, ${acc.targetName} $memberName")
-    sym
+    sym.orElse {
+      val originalName = acc.name.asTermName.originalOfSuperAccessorName
+      report.error(em"Member method ${originalName.debugString} of mixin ${acc.owner} is missing a concrete super implementation in $base.", base.srcPos)
+      acc
+    }
   }
 }

--- a/tests/neg/i11170.check
+++ b/tests/neg/i11170.check
@@ -1,0 +1,4 @@
+-- Error: tests/neg/i11170.scala:7:15 ----------------------------------------------------------------------------------
+7 |abstract class A extends U[String] // error
+  |               ^
+  |               Member method foo of mixin trait U is missing a concrete super implementation in class A.

--- a/tests/neg/i11170.scala
+++ b/tests/neg/i11170.scala
@@ -1,0 +1,8 @@
+trait T[X] {
+  def foo(x: X): X
+}
+trait U[X] extends T[X] {
+  abstract override def foo(x: X): X = super.foo(x)
+}
+abstract class A extends U[String] // error
+

--- a/tests/neg/i11170a.scala
+++ b/tests/neg/i11170a.scala
@@ -1,0 +1,28 @@
+package apackage {
+
+  abstract class A {
+    protected val x: Int
+  }
+  abstract class A2 {
+    val x: Int
+  }
+}
+
+package bpackage {
+  import apackage._
+
+  trait B extends A {
+    println(x)
+  }
+  trait B2 extends A2 {
+    println(x)
+  }
+}
+
+package cpackage {
+  import apackage._
+  import bpackage._
+
+  case class C(override protected val x: Int) extends A with B  // error
+  case class C2(override val x: Int) extends A2 with B2
+}

--- a/tests/pos/MathSpec.scala
+++ b/tests/pos/MathSpec.scala
@@ -1,0 +1,21 @@
+trait MathSig {
+  def sqrt(x: Double): Double
+}
+
+trait MathSpec extends MathSig {
+  val epsilon = 0.00001
+  abstract override def sqrt(x: Double) = {
+    require(x >= 0)
+    super.sqrt(x)
+  } ensuring { result =>
+    (x * x - result).abs < epsilon
+  }
+}
+
+trait MathImpl extends MathSig {
+  def sqrt(x: Double): Double =
+    ???
+}
+
+object Math extends MathImpl
+               with MathSpec

